ddtrace/opentelemetry/span.py

```diff
@@ -0,0 +1,125 @@
+import traceback
+from typing import TYPE_CHECKING
+
+from opentelemetry.trace import Span as Span
+from opentelemetry.trace import SpanContext
+from opentelemetry.trace import SpanKind
+from opentelemetry.trace import Status
+from opentelemetry.trace import StatusCode
+
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Dict
+    from typing import Optional
+    from typing import Type
+    from typing import Union
+
+    from opentelemetry.util.types import AttributeValue
+    from opentelemetry.util.types import Attributes
+
+    from ddtrace.span import Span as DDSpan
+
+
+class OtelSpan(Span):
+    """Implements the Opentelemetry Span API to create and configure datadog spans"""
+
+    def __init__(
+        self,
+        datadog_span,  # type: DDSpan
+        kind=SpanKind.INTERNAL,  # type: SpanKind
+        attributes=None,  # type: Optional[Attributes]
+        start_time=None,  # type: Optional[int]
+        record_exception=True,  # type: Optional[bool]
+        set_status_on_exception=True,  # type: Optional[bool]
+    ):
+        # (...) -> None
+        if start_time is not None:
+            datadog_span.start = start_time
+
+        # Only set if it isn't "internal" to save on bytes
+        if kind != SpanKind.INTERNAL:
+            # TODO: Validate/clean this?
+            datadog_span.set_tag_str("span.kind", kind.name.lower())
+
+        # TODO: Are there other default tags we want to set?
+
+        self._span = datadog_span
+        self._record_execption = record_exception
+        self._set_status_on_exception = set_status_on_exception
+
+        if attributes:
+            self.set_attributes(attributes)
+
+    def end(self, end_time=None):
+        # type: (Optional[int]) -> None
+        """Ends a datadog span"""
+        self._span.finish(finish_time=end_time)
+
+    def get_span_context(self):
+        # type: () -> SpanContext
+        """Returns an opentelemetry SpanContext using values from the active datadog span"""
+        # TODO: This is the bare minimum to get current context, but what about tracestate and flags?
+        return SpanContext(self._span.trace_id, self._span.span_id, False)
```

Comment:
is_remote_span will always be set to False here ([docs on otel remote spans](https://opentelemetry.io/docs/reference/specification/trace/api/#isremote)).

 Is this statement correct: A datadog span never be "remote" since we only propagate a context object across trace boundaries.